.github/workflows/motion-tests.yml

```diff
@@ -0,0 +1,78 @@
+name: Motion Extended Tests
+
+on:
+  workflow_dispatch:
+  workflow_call:
+
+jobs:
+  check-files:
+    name: Check files
+    runs-on: ubuntu-latest
+    outputs:
+      EXTENDED_MOTION_TEST: ${{ steps.file_checker.outputs.EXTENDED_MOTION_TEST }}
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+        with:
+          ref: ${{ github.event_name == 'pull_request_target' && github.event.pull_request.head.sha || github.ref }}
+          fetch-depth: 0
+      - name: check modified files
+        id: file_checker
+        run: |
+          ( git diff --name-only origin/${{github.base_ref}} | grep -e "motionplan" \
+            -e "referenceframe" \
+            -e "spatialmath" \
+            -e "components\/base\/wheeled" \
+            -e "components\/base\/kinematicbase" \
+            -e "services/motion/builtin" && echo "EXTENDED_MOTION_TEST=true" >> "$GITHUB_OUTPUT" ) || \
+            ( echo "EXTENDED_MOTION_TEST=false" >> "$GITHUB_OUTPUT" )
+
+  motion_tests:
+    name:  Motion Extended Tests
+    needs: check-files
+    runs-on: [buildjet-8vcpu-ubuntu-2204]
+    container: ghcr.io/viamrobotics/rdk-devenv:amd64-cache
+    timeout-minutes: 30
+    steps:
+
+    - name: Check out PR branch code
+      if: github.event_name == 'pull_request_target'
+      uses: actions/checkout@v3
+      with:
+        ref: ${{ github.event.pull_request.head.sha }}
+
+    - name: Check out code in motion-testing repository
+      uses: actions/checkout@v3
+      with:
+        repository: viamrobotics/motion-testing
+        path: motion-testing
+
+    - name: Change ownership to testbot
+      run: chown -R testbot:testbot .
+
+    - name: Run motion tests on PR branch
+      if: ${{ needs.check-files.outputs.EXTENDED_MOTION_TEST == 'true' }}
```

Comment:
why is the step conditional, rather than the entire job?

Reply:
Since the step is now required to pass to merge a PR, I did not want to explore the logic that would be necessary to make a step conditionally required rather than always required, and only conditionally running something difficult.

Reply:
Agree with Abe here, this needs to be conditional for the job itself. Otherwise this'll spool up an entire 8 core machine just to do nothing with it. Which in turn makes the job checking for file changes pointless as a separate job (it could just be a previous step if we're not saving the compute/spin-up time.)

It should be just a matter of checking that the result is "success" OR "skipped" in the test_passing job.

Reply:
I will take a stab at this and make sure everything still works